src/transaction/validation.rs

```diff
@@ -65,7 +88,22 @@ impl Output {
 }
 
 impl Transaction {
+    #[cfg(feature = "std")]
     pub fn validate(&self, block_height: Word) -> Result<(), ValidationError> {
+        self._validate(block_height, true)
+    }
+
+    #[cfg(feature = "std")]
+    pub fn validate_without_signature(&self, block_height: Word) -> Result<(), ValidationError> {
```

Comment:
Is there a use-case where we'd want to validate just the witnesses and nothing else?

Reply:
Yes: signatures can be validated in parallel, on a GPU even (see: Solana). In such a case, signature verification should be independent from everything else. The owner of each input is provided in the tx, so you can proceed optimistically even without verifying signatures.

Reply:
Added API to validate with or without signature:

https://github.com/FuelLabs/fuel-tx/pull/57/commits/a77f4e84eb40e75a806e03c41ce44285b6a18bdc